### PR TITLE
Workaround to prevent X10 crashes during benchmarks

### PR DESCRIPTION
**Pull request by BradLarson**
Larger X10 models occasionally cause crashes when run in our benchmarks, but these crashes were only occurring at the very end of the benchmark. Tracking this down, it appears to be due to X10 (at least on GPU) executing traces asynchronously, leading to them running slightly after the point that the model and optimizer have gone out of scope. Inserting a synchronous barrier that blocks until execution is finished prevents these crashes.

This is inserted immediately before the returns of the timing values, because I found that this is the only location that reliably prevents crashes. Placing this barrier in a `defer` doesn't achieve the desired results, and the barrier must be within the benchmark function where the model and optimizer are allocated.

This addresses issue #528, but a more robust fix may be needed at the API level.

### CODE REVIEW

File: Benchmarks/Models/ImageClassificationInference.swift [hunk 1]
```diff
@@ -65,10 +65,20 @@ where
                 batchTimings.append(batchTime)
                 currentBatch += 1
                 if case let .batches(batches) = duration, currentBatch >= batches {
+                    if backend == .x10 {
```
Comment (xihui-wu) [inline Benchmarks/Models/ImageClassificationInference.swift:68]: Should we add this to line 55 as well ?

File: Benchmarks/Models/ImageClassificationTraining.swift [hunk 2]
```diff
@@ -74,11 +74,21 @@ where
                 batchTimings.append(durationInMilliseconds(since: beforeBatch))
                 currentBatch += 1
                 if case let .batches(batches) = duration, currentBatch >= batches {
+                    if backend == .x10 {
```
Comment (xihui-wu) [inline Benchmarks/Models/ImageClassificationTraining.swift:77]: Same, should add before line 62?

File: Benchmarks/Models/ImageClassificationInference.swift (hunk 1 above)
Comment (BradLarson) [reply Benchmarks/Models/ImageClassificationInference.swift:73]: Good catch. I've added the same above.

File: Benchmarks/Models/ImageClassificationTraining.swift (hunk 2 above)
Comment (BradLarson) [reply Benchmarks/Models/ImageClassificationTraining.swift:82]: Likewise, this has been added above. Thanks for pointing it out.

**Pull request merged**